### PR TITLE
fix build and add functions to dll

### PR DESCRIPTION
**Pull request by Simplxss**

### CODE REVIEW

**Review by ihedvall — APPROVED** [review]

**Review by ihedvall — COMMENTED** [review]
Hint: There is a std::string::data() function added so you can write something like:

```
std::string s(size);
MdfTagGetName(eTag, s.data());
```

instead

**Comment by Simplxss** [comment]
> Hint: There is a std::string::data() function added so you can write something like:
> 
> ```
> std::string s(size);
> MdfTagGetName(eTag, s.data());
> ```
> 
> instead

It doesn't work. std::string is not std::vector.

The following code works sometimes, but sometimes it doesn't,
```cpp
    std::string str;
    str.reserve(MdfAttachmentGetFileType(attachment, nullptr) + 1);
    str.resize(MdfAttachmentGetFileType(attachment, str.data()));
    return str;
```

**Comment by ihedvall** [comment]
std::string s(size, '\0'); ?

File: mdflibrary/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+
 message(STATUS  "Compiler Version " ${CMAKE_CXX_COMPILER_VERSION})
 message(STATUS  "MSVC " ${MSVC})
-if (MSVC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.36)
+if (MDF_BUILD_SHARED_LIB_NET)
     message(STATUS  "CLR compile")
     add_library(mdflibrary SHARED
-            src/MdfExport.cpp src/MdfExport.h
+            # C++ files
+            src/MdfExport.cpp
```
Comment (ihedvall) [inline mdflibrary/CMakeLists.txt:18]: According to the CMake manual, you should add library as a MODULE. If so, you dont't need the Export.cpp file. The -NET have all export function. Well worth a try at least

**Review by ihedvall — APPROVED** [review]
Minor check if the SHARED ->MODULE is a simpler solution

File: mdflibrary/CMakeLists.txt (hunk 1 above)
Comment (Simplxss) [reply mdflibrary/CMakeLists.txt:18]: Cannot. `src/MdfExport.cpp` contains functions that need to be exported. If change SHARED to MODULE, cmake would not export symbols to dll.
As the CMake manual said.
"For example, a Windows resource DLL or a managed C++/CLI DLL that ***exports no unmanaged symbols*** would need to be a MODULE library."

Comment (Simplxss) [reply mdflibrary/CMakeLists.txt:18]: I trid to change SHARED to MODULE, and cmake complain that `Target "mdflibrary" of type MODULE_LIBRARY may not be linked into another target. One may link only to INTERFACE, OBJECT, STATIC or SHARED libraries, or to executables with the ENABLE_EXPORTS property set.`

**Pull request merged**